skops/hub_utils/_hf_hub.py

```diff
@@ -415,19 +418,64 @@ def init(
     ):
         requirements.append("scikit-learn-intelex")
 
+    # model parameter can be either a path or a model object
     try:
-        shutil.copy2(src=model, dst=dst)
-
-        model_name = model.name
-        _create_config(
-            model_path=model_name,
-            requirements=requirements,
-            dst=dst,
-            task=task,
-            data=data,
-            model_format=model_format,
-            use_intelex=use_intelex,
-        )
+        if isinstance(model, (str, Path)):
+            model = _check_model_file(model)
+            shutil.copy2(src=model, dst=dst)
+
+            model_name = model.name
+
+            _create_config(
+                model_path=model_name,
+                requirements=requirements,
+                dst=dst,
+                task=task,
+                data=data,
+                model_format=model_format,
+                use_intelex=use_intelex,
+            )
+
+            # load model from file
+            model = _load_model(model, trusted=True)
+            model_card = card.Card(model, metadata=card.metadata_from_config(dst))
+            model_card.save(dst / "README.md")
+        elif isinstance(model, BaseEstimator):
```

Comment:
I think we need no check for `BaseEstimator` here, so this can just be `else` and the error below can be removed. There could be valid models here that don't inherit from `BaseEstimator`. It is the user's responsibility to provide an sklearn-compatible model.

Reply:
Okay, got it!